lib/puzzles/cubit/puzzles_cubit.dart

```diff
@@ -0,0 +1,163 @@
+import 'package:custom_platform_device_id/platform_device_id.dart';
+import 'package:equatable/equatable.dart';
+import 'package:hydrated_bloc/hydrated_bloc.dart';
+import 'package:json_annotation/json_annotation.dart';
+import 'package:puzzle_repository/puzzle_repository.dart';
+
+part 'puzzles_state.dart';
+part 'puzzles_cubit.g.dart';
+
+class PuzzlesCubit extends HydratedCubit<PuzzlesState> {
+  PuzzlesCubit({required PuzzleRepository puzzleRepository})
+      : _puzzleRepository = puzzleRepository,
+        super(const PuzzlesState());
+
+  final PuzzleRepository _puzzleRepository;
+
+  Future<void> verifyUser({required String username}) async {
+    try {
+      emit(
+        state.copyWith(
+          userVerificationStatus: UserVerificationStatus.verifying,
+        ),
+      );
+      final deviceId = await PlatformDeviceId.getDeviceId;
```

Comment:
This change will require me to change the privacy policy of the app since we currently do no collect any device information. Why is this needed?

Reply:
Hey @felangel , Since we don't have any additional authentication methods, we're binding the username to the `device ID`. For example, if one user chooses the username `john` and another user also selects the same name, we want to be able to differentiate between them based on their `device ID`.

Reply:
Hey @felangel The issue with using UUIDs is that they need to be cached. If a user uninstalls the app and later tries to access the puzzle again, they won't be able to continue their progress and will have to start as a new user. Anyways i have updated the code to use uuid.

Reply:
Yeah that's okay though, I think the uninstallation case is fine since users likely won't be installing and uninstalling the app during the conference. Most people will install it once before or during the conference and then uninstall it after the conference. Thanks for making the change!